api/src/main/java/org/pmiops/workbench/cdr/dao/CriteriaDao.java

```diff
@@ -9,7 +9,13 @@
 
 public interface CriteriaDao extends CrudRepository<Criteria, Long> {
 
-    @Query("select c from Criteria c where c.type like :type% and c.parentId = :parentId order by c.code asc")
-    List<Criteria> findCriteriaByTypeAndParentId(@Param("type") String type, @Param("parentId") Long parentId);
+    List<Criteria> findCriteriaByTypeAndParentIdOrderByCodeAsc(@Param("type") String type, @Param("parentId") Long parentId);
+
+    @Query("select c from Criteria c " +
+            "where c.type like :type " +
+            "and (lower(c.name) like lower(:name) or lower(c.code) like lower(:code)) " +
+            "and c.selectable = 1 " +
+            "order by c.code asc")
+    List<Criteria> findCriteriaByTypeAndNameOrCode(@Param("type") String type, @Param("name") String name, @Param("code") String code);
```

Comment:
Why are name and code separate parameters here? Isn't there going to be just a single text box? Isn't querying by just name (what's actually visible) what we want anyway?

Reply:
Corrent. Oversight by me. I was using the query by method name in which you have to use 2 different variables but have switched to using the @Query annotation. So i no longer need 2 variables.